packages/notebook/src/browser/view/notebook-cell-list-view.tsx

```diff
@@ -133,32 +152,36 @@ export class NotebookCellListView extends React.Component<CellListProps, Noteboo
     }
 
     protected onAddNewCell(kind: CellKind, index: number): void {
-        this.props.commandRegistry.executeCommand(NotebookCommands.ADD_NEW_CELL_COMMAND.id,
-            this.props.notebookModel,
-            kind,
-            index
-        );
+        if (this.isEnabled()) {
+            this.props.commandRegistry.executeCommand(NotebookCommands.ADD_NEW_CELL_COMMAND.id,
+                this.props.notebookModel,
+                kind,
+                index
+            );
+        }
     }
 
     protected shouldRenderDragOverIndicator(cell: NotebookCellModel, position: 'top' | 'bottom'): boolean {
-        return this.state.dragOverIndicator !== undefined &&
+        return this.isEnabled() &&
+            this.state.dragOverIndicator !== undefined &&
             this.state.dragOverIndicator.cell === cell &&
             this.state.dragOverIndicator.position === position;
     }
 
 }
 
 export interface NotebookCellDividerProps {
+    isVisible: () => boolean;
     onAddNewCell: (type: CellKind) => void;
     onDrop: (event: React.DragEvent<HTMLLIElement>) => void;
     onDragOver: (event: React.DragEvent<HTMLLIElement>) => void;
 }
 
-export function NotebookCellDivider({ onAddNewCell, onDrop, onDragOver }: NotebookCellDividerProps): React.JSX.Element {
+export function NotebookCellDivider({ isVisible, onAddNewCell, onDrop, onDragOver }: NotebookCellDividerProps): React.JSX.Element {
     const [hover, setHover] = React.useState(false);
 
     return <li className='theia-notebook-cell-divider' onMouseEnter={() => setHover(true)} onMouseLeave={() => setHover(false)} onDrop={onDrop} onDragOver={onDragOver}>
```

Comment:
We need the `isEnabled()` check anyway, since you can also drag on top of a cell. I'd love to simplify this, but probably not in the scope of this PR.